src/code_807E5AC.c

```diff
@@ -0,0 +1,1498 @@
+            if (AbilityIsActive(pokemon, ABILITY_FORECAST)) {
+                info->apparentID = GetMonsterApparentID(pokemon, info->id);
+            }
+            else {
+                info->apparentID = MONSTER_CASTFORM;
+            }
+            if (info->apparentID != previousApparentId) {
+                pokemon->axObj.spriteFile = GetSpriteData(info->apparentID);
+                sub_806CCB4(pokemon,sub_806CEBC(pokemon));
+                flag = TRUE;
+            }
+        }
+        if (flag) {
+            sub_80429D8(pokemon);
+        }
+    }
+}
+
+void GetWeatherName(u8 *buffer, u8 weather)
+{
+    strcpy(buffer, gWeatherNames[weather]);
+}
+
+void MudWaterSportEffect(u8 which)
+{
+    if (which == 0) {
+        gDungeon->weather.mudSportTurns = CalculateStatusTurns(0, gMudWaterSportTurnRange, FALSE);
+        LogMessageByIdWithPopupCheckUser(GetLeader(), gUnknown_80FAD8C);
+    }
+    else {
+        gDungeon->weather.waterSportTurns =CalculateStatusTurns(0, gMudWaterSportTurnRange, FALSE);
+        LogMessageByIdWithPopupCheckUser(GetLeader(), gUnknown_80FADB0);
+    }
+}
+
+bool8 sub_807EAA0(bool8 arg0, bool8 arg1)
+{
+    s32 i;
+    s32 var = 0;
+    s32 index = -1;
+    bool8 ret = FALSE;
+    s32 weatherBefore = GetApparentWeather(NULL);
+
+    for (i = 0; i < 8; i++) {
+        if (gDungeon->weather.unkE267[i] < gDungeon->weather.naturalWeather[i]) {
+            gDungeon->weather.unkE267[i] = gDungeon->weather.naturalWeather[i];
+        }
+        else if (gDungeon->weather.unkE267[i] > gDungeon->weather.naturalWeather[i]) {
+            sub_80838EC(&gDungeon->weather.unkE267[i]);
+        }
+
+        if (var < gDungeon->weather.unkE267[i]) {
+            var = gDungeon->weather.unkE267[i];
+            index = i;
+        }
+    }
+
+    if (index < 0) {
+        gDungeon->weather.weather = gDungeon->weather.unkE265;
+    }
+    else {
+        gDungeon->weather.weather = index;
+    }
+
+    if (gDungeon->weather.nullifyWeather) {
+        gDungeon->weather.weather = 0;
+    }
+
+    if (weatherBefore != GetApparentWeather(NULL) || arg1) {
+        ret = TRUE;
+        sub_807E698(arg0);
+    }
+
+    if (gDungeon->weather.weatherDamageCounter != 0) {
+        gDungeon->weather.weatherDamageCounter--;
+    }
+    else {
+        gDungeon->weather.weatherDamageCounter = 9;
+    }
+
+    if (gDungeon->weather.mudSportTurns != 0) {
+        sub_80838EC(&gDungeon->weather.mudSportTurns);
+        if (gDungeon->weather.mudSportTurns == 0) {
+            LogMessageByIdWithPopupCheckUser(GetLeader(), gUnknown_80FA778);
+        }
+    }
+    if (gDungeon->weather.waterSportTurns != 0) {
+        sub_80838EC(&gDungeon->weather.waterSportTurns);
+        if (gDungeon->weather.waterSportTurns == 0) {
+            LogMessageByIdWithPopupCheckUser(GetLeader(), gUnknown_80FA758);
+        }
+    }
+
+    return ret;
+}
+
+void sub_807EC28(bool8 arg0)
```

Comment:
Should we label these few funcs with some name mentioning that this is Dungeon Kecleon shop.